solutions/cf-616D.mdx

```diff
@@ -0,0 +1,79 @@
+---
+id: cf-616D
+source: CF
+title: Longest k-Good Segment
+author: Sofia Yang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/22712)
+
+<LanguageSection>
+
+<JavaSection>
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+## Java Implementation
+
+```java
+public class kgood {
+	CodeSnip{Kattio}
+
+    public static int MAX = (int) 1e6;
+
+    public static void main(String[] args) throws IOException {
+        Kattio io = new Kattio();
+        int N = io.nextInt();
+        int K = io.nextInt();
+
+        int[] A = new int[MAX];
```

Comment:
it'd be cool if you could
1. replace `MAX` with `N` (i mean arrays in java don't have to constant size)
2. rename this to `arr` (just a tiny bit more explicit)